components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/dao/AuthenticatorManagementDAO.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.common.dao;
+
+import org.wso2.carbon.identity.application.common.exception.AuthenticatorMgtException;
+import org.wso2.carbon.identity.application.common.model.UserDefinedLocalAuthenticatorConfig;
+import org.wso2.carbon.identity.base.AuthenticatorPropertyConstants.AuthenticationType;
+
+import java.util.List;
+
+/**
+ * This interface performs CRUD operations for the User defined Local Application Authenticator configurations.
+ */
+public interface AuthenticatorManagementDAO {
+
+    /**
+     * Create a new user defined Local Application Authenticator configuration.
+     *
+     * @param authenticatorConfig   Local Application Authenticator configuration.
+     * @param tenantId              Tenant Id.
+     *
+     * @return Created UserDefinedLocalAuthenticatorConfig.
+     * @throws AuthenticatorMgtException If an error occurs while adding the authenticator configuration.
+     */
+    UserDefinedLocalAuthenticatorConfig addUserDefinedLocalAuthenticator(
+            UserDefinedLocalAuthenticatorConfig authenticatorConfig, int tenantId, AuthenticationType type)
+            throws AuthenticatorMgtException;
```

Comment:
I'm actually thinking at the DAO layer we should ideally have an exception type that is different from the service layer which incorporates error codes and messages. However this seems to be the practice we are following. So it's fine to proceed, but some are we need to think through

Reply:
@ashanthamara FYI